.pipelines/build-windows-nuget.yml

```diff
@@ -34,6 +44,11 @@ steps:
   env:
     nugetPath: $(Build.SourcesDirectory)\vowpalwabbit\.nuget\nuget.exe
   failOnStderr: true
+- task: Cache@2
+  inputs:
+    key: '$(Build.SourcesDirectory)\cs\**\*.csproj'
+    path: $(Build.SourcesDirectory)\vowpalwabbit\out
```

Comment:
Why cache the output? Is this for the obj files? Does it improve things?

Reply:
it might improve things for a little bit since I'm seeing on the log ```0 of 3673 functions ( 0.0%) were compiled, the rest were copied from previous compilation.```

Reply:
Hopefully it is able to reliably determine when to recompile